dynawo/sources/Models/CPP/ModelNetwork/DYNModelCurrentLimits.cpp

```diff
@@ -57,60 +58,42 @@ ModelCurrentLimits::setMaxTimeOperation(const double& maxTimeOperation) {
   maxTimeOperation_ = maxTimeOperation;
 }
 
-void
-ModelCurrentLimits::setNbLimits(const int& number) {
-  nbLimits_ = number;
-  activated_.assign(number, false);
-  tLimitReached_.assign(number, std::numeric_limits<double>::quiet_NaN());
-}
-
 void
 ModelCurrentLimits::addLimit(const double& limit) {
-  limits_.push_back(limit);
-  if (std::isnan(limit))
-    limitActivated_.push_back(false);
-  else
-    limitActivated_.push_back(true);
+  if (!std::isnan(limit)) {
+    limits_.push_back(limit);
+    activated_.push_back(false);
+    tLimitReached_.push_back(std::numeric_limits<double>::quiet_NaN());
+    nbLimits_++;
```

Comment:
nbLimits == limits_.size() so I don't think that an a attribute is required here